src/pages/Search/components/Heatmap.js

```diff
@@ -12,36 +16,56 @@ import {
   BoxRow,
   TimeZone,
 } from './Heatmap.styles';
-import Box from './Box';
 
 const Heatmap = ({ posts }) => {
-  const rows = posts.map((el) => {
-    const columns = el.map((totalPosts) => (
-      <Box numTotalPosts={totalPosts.length} />
-    ));
-    return <BoxRow key={el}>{columns}</BoxRow>;
+  const [selected, setSelected] = useState('');
+  const [timePosts, setTimePosts] = useState([]);
```

Comment:
This causes a duplication of items, thus you won't have a [single source of truth](http://www.hackingwithreact.com/read/1/12/state-and-the-single-source-of-truth). You can simply store the day and hour inside `selected` when you use an object there. The `selected` state would look like this { day: 0, hour: 15 }

Reply:
This is still open. Not sure if I explained myself well. If you have questions just leave a comment

Reply:
I don't understand, the `selected `state should have day and hour, right?

Reply:
do you mean I should use 1 state and store posts and day, an hour?

Reply:
Yes, you should only use the `selected` state. And no, you shouldn't store the posts inside the state. The problem is that you already have all the posts inside the `posts` prop, right? When you have another state `timePosts` you only copy some of the objects from `posts` to `timePosts`. So they are duplicated. This can cause problems with syncing and outdated data. That's what I was referring to with single source of truth.

What you can do is filter the `posts` array. So when you have `selected.day` and `selected.hour` (which are both indeces of the posts array, right?) you can simply get the selected posts by doing this in your render function

```
const selectedPosts = selected.day !== undefined ? posts[selected.day][selected.hour] : []
```  

Was this explanation better?

Reply:
Yes, thank you